persistent/Database/Persist/Types/Base.hs

```diff
@@ -654,9 +652,15 @@ instance A.ToJSON PersistValue where
     toJSON PersistNull = A.Null
     toJSON (PersistList l) = A.Array $ V.fromList $ map A.toJSON l
     toJSON (PersistMap m) = A.object $ map (second A.toJSON) m
-    toJSON (PersistDbSpecific b) = A.String $ T.cons 'p' $ TE.decodeUtf8 $ B64.encode b
-    toJSON (PersistLiteral b) = A.String $ T.cons 'l' $ TE.decodeUtf8 $ B64.encode b
-    toJSON (PersistLiteralEscaped b) = A.String $ T.cons 'e' $ TE.decodeUtf8 $ B64.encode b
+    toJSON (PersistLiteral_ litTy b) =
+        let encoded = TE.decodeUtf8 $ B64.encode b
+            prefix =
+                case litTy of
+                    DbSpecific -> 'p'
+                    Unescaped -> 'l'
+                    Escaped -> 'e'
+         in
+            A.String $ T.cons prefix encoded
```

Comment:
This is *exactly* the problem I was trying to solve, and *exactly* the problem that my patch introduced. Sigh. I'm disappointed that I didn't catch it.

So - the patch makes `PersistDbSpecific` a catch-all match, expanding to `PersistLiteral_ _ bs`. Which means that the `PersistDbSpecific` woudl encode *all* values with the `p` prefix! That's bad. This fixes that problem.